integration-testing/tests/users/set-dating-status.test.js

```diff
@@ -0,0 +1,144 @@
+        status: 'ENABLED',
+      },
+    }),
+  ).rejects.toThrow(/ClientError: `fullName` is required field/)
+
+  // Set fullName
+  await ourClient.mutate({mutation: mutations.setUserDetails, variables: {fullName: 'Hunter S'}})
+  await expect(
+    ourClient.mutate({
+      mutation: mutations.setUserDatingStatus,
+      variables: {
+        status: 'ENABLED',
+      },
+    }),
+  ).rejects.toThrow(/ClientError: `photoPostId` is required field/)
+
+  // Set photoPostId
+  const postId = uuidv4()
+  await ourClient
+    .mutate({mutation: mutations.addPost, variables: {postId, imageData: grantDataB64, takenInReal: true}})
+    .then(({data: {addPost: post}}) => expect(post.postId).toBe(postId))
+  await ourClient
+    .mutate({mutation: mutations.setUserDetails, variables: {photoPostId: postId}})
+    .then(({data: {setUserDetails: user}}) => expect(user.photo.url).toBeTruthy())
+
+  await expect(
+    ourClient.mutate({
+      mutation: mutations.setUserDatingStatus,
+      variables: {
+        status: 'ENABLED',
+      },
+    }),
+  ).rejects.toThrow(/ClientError: `gender` is required field/)
+
+  // Set gender
+  await ourClient.mutate({mutation: mutations.setUserDetails, variables: {gender: 'MALE'}})
+  await expect(
+    ourClient.mutate({
+      mutation: mutations.setUserDatingStatus,
+      variables: {
+        status: 'ENABLED',
+      },
+    }),
+  ).rejects.toThrow(/ClientError: `currentLocation` is required field/)
+
+  // Set currentLocation
+  await ourClient.mutate({
+    mutation: mutations.setUserCurrentLocation,
+    variables: {
+      latitude: 50.01,
+      longitude: 50.01,
+      accuracy: 20,
+    },
+  })
+  await expect(
+    ourClient.mutate({
+      mutation: mutations.setUserDatingStatus,
+      variables: {
+        status: 'ENABLED',
+      },
+    }),
+  ).rejects.toThrow(/ClientError: `matchGenders` is required field/)
+
+  // Set matchGenders
+  await ourClient.mutate({
+    mutation: mutations.setUserDetails,
+    variables: {
+      matchGenders: ['MALE', 'FEMALE'],
+    },
+  })
+  await expect(
+    ourClient.mutate({
+      mutation: mutations.setUserDatingStatus,
+      variables: {
+        status: 'ENABLED',
+      },
+    }),
+  ).rejects.toThrow(/ClientError: `matchAgeRange` is required field/)
+
+  // Set matchAgeRange
+  await ourClient.mutate({
+    mutation: mutations.setUserAgeRange,
+    variables: {
+      min: 20,
+      max: 50,
+    },
+  })
+  await expect(
+    ourClient.mutate({
+      mutation: mutations.setUserDatingStatus,
+      variables: {
+        status: 'ENABLED',
+      },
+    }),
+  ).rejects.toThrow(/ClientError: `matchLocationRadius` is required field/)
+
+  // Set matchLocationRadius
+  await ourClient.mutate({
+    mutation: mutations.setUserDetails,
+    variables: {
+      matchLocationRadius: 20,
+    },
+  })
+  // All required fields are fulfilled
+  await ourClient.mutate({mutation: mutations.setUserDatingStatus, variables: {status: 'ENABLED'}})
```

Comment:
```suggestion
  await ourClient
    .mutate({mutation: mutations.setUserDatingStatus, variables: {status: 'ENABLED'}})
    .then(({data: {setUserDatingStatus: user}}) => expect(user.datingStatus).toBe('ENABLED'))
```

(apologies if there are typos in this suggestion, I did not actually run it